opsdroid/tests/test_connector.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+import asynctest.mock as amock
+
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message, Reaction
+
+
+class TestConnectorBaseClass:
+    """Test the opsdroid connector base class."""
+
+    def test_init(self, opsdroid):
+        config = {"example_item": "test"}
+        connector = Connector(config, opsdroid=opsdroid)
+        assert connector.default_target is None
+        assert connector.name == ""
+        assert connector.config["example_item"] == "test"
+
+    def test_property(self):
+        opsdroid = amock.CoroutineMock()
+        connector = Connector({"name": "shell"}, opsdroid=opsdroid)
+        assert connector.configuration.get("name") == "shell"
+
+    def test_connect(self, event_loop, no_config_connector):
+        with pytest.raises(NotImplementedError):
+            event_loop.run_until_complete(no_config_connector.connect())
+
+    def test_listen(self, event_loop, no_config_connector):
+        with pytest.raises(NotImplementedError):
+            event_loop.run_until_complete(no_config_connector.listen())
+
+    def test_respond(self, event_loop, no_config_connector):
+        with pytest.raises(TypeError):
+            event_loop.run_until_complete(no_config_connector.send(Message("")))
+
+    def test_unsupported_event(self, event_loop, no_config_connector):
+        with pytest.raises(TypeError):
+            event_loop.run_until_complete(no_config_connector.send(Reaction("emoji")))
+
+    def test_incorrect_event(self):
+        class NotanEvent:
+            pass
+
+        class MyConnector(Connector):
+            @register_event(NotanEvent)
+            def send_my_event(self, event):
+                pass
+
+        with pytest.raises(TypeError):
+            MyConnector()
+
+    def test_event_subclasses(self):
+        class MyEvent(Message):
+            pass
+
+        class MyConnector(Connector):
+            @register_event(Message, include_subclasses=True)
+            def send_my_event(self, event):
+                pass
+
+        c = MyConnector({})
```

Comment:
Could we turn this `c` variable into something else like `connector?`

Reply:
Changed it.